x-pack/plugins/reporting/server/browsers/chromium/driver_factory/start_logs.ts

```diff
@@ -0,0 +1,125 @@
+
+const browserLaunchTimeToWait = 5 * 1000;
+
+// Default args used by pptr
+// https://github.com/puppeteer/puppeteer/blob/13ea347/src/node/Launcher.ts#L168
+const defaultArgs = [
+  '--disable-background-networking',
+  '--enable-features=NetworkService,NetworkServiceInProcess',
+  '--disable-background-timer-throttling',
+  '--disable-backgrounding-occluded-windows',
+  '--disable-breakpad',
+  '--disable-client-side-phishing-detection',
+  '--disable-component-extensions-with-background-pages',
+  '--disable-default-apps',
+  '--disable-dev-shm-usage',
+  '--disable-extensions',
+  '--disable-features=TranslateUI',
+  '--disable-hang-monitor',
+  '--disable-ipc-flooding-protection',
+  '--disable-popup-blocking',
+  '--disable-prompt-on-repost',
+  '--disable-renderer-backgrounding',
+  '--disable-sync',
+  '--force-color-profile=srgb',
+  '--metrics-recording-only',
+  '--no-first-run',
+  '--enable-automation',
+  '--password-store=basic',
+  '--use-mock-keychain',
+  '--remote-debugging-port=0',
+  '--headless',
+];
+
+export const browserStartLogs = (
+  core: ReportingCore,
+  logger: LevelLogger,
+  overrideFlags: string[] = []
+) => {
+  const config = core.getConfig();
+  const proxy = config.get('capture', 'browser', 'chromium', 'proxy');
+  const disableSandbox = config.get('capture', 'browser', 'chromium', 'disableSandbox');
+  const userDataDir = mkdtempSync(join(tmpdir(), 'chromium-'));
+  const binaryPath = getBinaryPath();
+  const kbnArgs = args({
+    userDataDir,
+    viewport: { width: 800, height: 600 },
+    disableSandbox,
+    proxy,
+  });
+  const finalArgs = uniq([...defaultArgs, ...kbnArgs, ...overrideFlags]);
+
+  // On non-windows platforms, `detached: true` makes child process a
+  // leader of a new process group, making it possible to kill child
+  // process tree with `.kill(-pid)` command. @see
+  // https://nodejs.org/api/child_process.html#child_process_options_detached
+  const browserProcess = spawn(binaryPath, finalArgs, {
+    detached: process.platform !== 'win32',
+  });
+
+  const rl = createInterface({ input: browserProcess.stderr });
+
+  const exit$ = fromEvent(browserProcess, 'exit').pipe(
+    map((code) => {
+      logger.error(`Browser exited abnormally, received code: ${code}`);
+      return i18n.translate('xpack.reporting.diagnostic.browserCrashed', {
+        defaultMessage: `Browser exited abnormally during startup`,
+      });
+    })
+  );
+
+  const error$ = fromEvent(browserProcess, 'error').pipe(
+    map(() => {
+      logger.error(`Browser process threw an error on startup`);
+      return i18n.translate('xpack.reporting.diagnostic.browserErrored', {
+        defaultMessage: `Browser process threw an error on startup`,
+      });
+    })
+  );
+
+  // Some chromium errors don't show up until a few seconds after startup
+  // hence why we wait for at least 5 seconds to let them surface
+  const log$ = fromEvent(rl, 'line').pipe(tap((log) => logger.info(`Chromium log "${log}"`)));
```

Comment:
I hate to be picky, but I would like to see this teased out a bit more. What about:

```suggestion
  const processLogger = logger.clone(['chromium-stderr']);
    const log$ = fromEvent(rl, 'line').pipe(
    tap((message: unknown) => {
      if (typeof message === 'string') {
        processLogger.info(message);
      }
    })
  );
```